### PR TITLE
Updated the content view resume task

### PR DESCRIPTION
**Pull request by devendra104**

### CODE REVIEW

File: upgrade/helpers/tasks.py [hunk 1]
```diff
@@ -573,6 +577,7 @@ def sync_tools_repos_to_upgrade(client_os, hosts):
                     host=host)
         else:
             host = entities.Host().search(query={'search': 'name={}'.format(host)})[0]
+            logger.info(f"Adding the Subscription {sub.name} on {host}")
```
Comment (jyejare) [inline upgrade/helpers/tasks.py:580]:
```suggestion
            logger.info(f"Adding the Subscription {sub.name} on {host=}")
```

**Review by jyejare — COMMENTED** [review]
ACK with non blocking comments.

**Review by jyejare — APPROVED** [review]

File: upgrade/helpers/tasks.py [hunk 2]
```diff
@@ -148,7 +148,9 @@ def sync_capsule_repos_to_upgrade(capsules):
         # Fix of 1770940, 1773601
         logger.info("Resuming the cancelled content view {} publish task"
                     .format(cv.name))
-        output = run("sleep 100; hammer task list|awk '/Task canceled/{print $1}'; sleep 100")
+        output = run("hammer task list|awk '/Task canceled/ && /paused/ {print $1}'")
+        for task_id in output.split():
+            run(f"sleep 100; hammer task resume --task-ids {task_id}; sleep 100")
```
[the remark this answers is not in the document]
Comment (devendra104) [reply upgrade/helpers/tasks.py]: yes, we can pass the entire tasks comma-separated. Earlier release I faced the problem of "no task is available" for resume and task is in-progress after the finish, that's the reason for keeping the 100 seconds gap before the start and after the finish. Now I have removed it and added a new function where we handle these things.

File: upgrade/helpers/tasks.py [hunk 3]
```diff
@@ -148,12 +148,7 @@ def sync_capsule_repos_to_upgrade(capsules):
         # Fix of 1770940, 1773601
         logger.info("Resuming the cancelled content view {} publish task"
                     .format(cv.name))
-        output = run("sleep 100; hammer task list|awk '/Task canceled/{print $1}'; sleep 100")
```
Comment (jameerpathan111) [inline upgrade/helpers/tasks.py:151]: @devendra104 there is a foreman-maintain check to resume paused tasks. Should we give it try? and see if that works for us here.

Comment (devendra104) [reply upgrade/helpers/tasks.py:151]: Thanks, Jameer for the review,  As we know,  foreman-maintain calls the same hammer command in the background to resume the task, I think to add one more layer is not good here.

**Review by jameerpathan111 — APPROVED** [review]
ACK

File: upgrade/helpers/tasks.py (hunk 1 above)
Comment (devendra104) [reply upgrade/helpers/tasks.py]: Faced some issue with Travis, so I rolldcback the changes

File: upgrade/helpers/tasks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (devendra104) [reply upgrade/helpers/tasks.py]: For now, I have removed the time gap, and tested with the standalone job and didn't face any issue, but in the future, If I will face the same problem, I will roll back the changes.

File: upgrade/helpers/tasks.py [hunk 4]
```diff
@@ -1284,3 +1272,19 @@ def job_execution_time(task_name, start_time=None):
...
         return start_time
+
+
+def resume_failed_task():
+    """
+    This function is used to resume the canceled paused tasks.
+    """
+    output = run("hammer task list|awk '/Task canceled/ && /paused/ {print $1}'")
+    run(f"hammer task resume --task-ids {','.join(output.split())}")
+    logger.info(f"The CV publish task {','.join(output.split())} has resumed successfully, "
```
Comment (jyejare) [inline upgrade/helpers/tasks.py:1283]: If this is an universal function to resume the paused tasks, there is no point in logging the CV Publish here. You can do this logging after calling this function in a code.

File: upgrade/helpers/tasks.py [hunk 5]
```diff
@@ -1284,3 +1272,19 @@ def job_execution_time(task_name, start_time=None):
...
+                f"waiting for their completion")
+    for task_id in output.split():
+        run(f'hammer task progress --id {task_id}')
```
Comment (jyejare) [inline upgrade/helpers/tasks.py:1286]: Why do you want to see the progress ? Is it for monitoring the job ? Also, is this command is `async` which wont hold ur program until the progress bar completes for each task?

**Review by jyejare — COMMENTED** [review]
Some comments for improvement !

File: upgrade/helpers/tasks.py (hunk 4 above)
Comment (devendra104) [reply upgrade/helpers/tasks.py:1283]: yes, you are right, I missed it.

File: upgrade/helpers/tasks.py (hunk 5 above)
Comment (devendra104) [reply upgrade/helpers/tasks.py:1286]: Some of the further tasks depend on their completion(on dependent cv task), so it is required, This command is async.

**Comment by devendra104** [comment]
@jyejare I have added your suggestion, please review it again.

**Pull request merged**